### PR TITLE
ForceRenderer - bold unit commanders

### LINKED ISSUE #5019: [RFE] Highlight Force Commanders in TO&E Tree View
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the RFE to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] Does this conform with our Content Policy linked above?
- [X] I've searched the GitHub tracker and haven't found a similar feature request listed

### RFE Details *

This is an improvement to an existing feature

### Brief Description *

Bold or otherwise highlight the commander of each force in the treeview of the TO&E:

Looks like it would involve this bit of code but I have no idea how to determine if the person is the group's commander.

https://github.com/WeaverThree/mekhq/blob/0e47eecb5c8b06f6e3e9b743832d29df155b8073/MekHQ/src/mekhq/gui/ForceRenderer.java#L61

### MekHQ Suite Version *

v0.50.1-SNAPSHOT: 2024-10-04

### Attach Files

_No response_

### Final Checklist

- [X] I've checked to make sure that this RFE has not already been filed
- [X] I am requesting implementation of only one relevant set of features in this ticket for clarity and focus

### PR DESCRIPTION
**Pull request by WeaverThree**
Closes #5019 - also cleans up the file a little though I'm not up for a major refactor today

### CODE REVIEW

File: MekHQ/src/mekhq/gui/ForceRenderer.java [hunk 1]
```diff
@@ -57,31 +57,29 @@ public Component getTreeCellRendererComponent(JTree tree, Object value, boolean
...
+            Person person = unit.getCommander();
             if (person != null) {
                 name = person.getFullTitle();
-                name += " (" + u.getEntity().getCrew().getGunnery() + '/'
-                        + u.getEntity().getCrew().getPiloting() + ')';
-                if (person.needsFixing() || (u.getEntity().getCrew().getHits() > 0)) {
+                name += " (" + unit.getEntity().getCrew().getGunnery() + '/'
+                        + unit.getEntity().getCrew().getPiloting() + ')';
+                if (person.needsFixing() || (unit.getEntity().getCrew().getHits() > 0)) {
                     name = "<font color='" + MekHQ.getMHQOptions().getFontColorNegativeHexColor() + "'>" + name
```
Comment (IllianiCBT) [inline MekHQ/src/mekhq/gui/ForceRenderer.java:67]: I know this isn't your code, but let's get the strings like this replaced with `ReportingUtilities.spanOpeningWithCustomColor(MekHQ.getMHQOptions().getFontColorNegativeHexColor())` and `ReportingUtilities.CLOSING_SPAN_TAG`

File: MekHQ/src/mekhq/gui/ForceRenderer.java [hunk 2]
```diff
@@ -57,31 +57,29 @@ public Component getTreeCellRendererComponent(JTree tree, Object value, boolean
...
                             + "</font>";
                 }
             }
-            String uname = "<i>" + u.getName() + "</i>";
-            if (u.isDamaged()) {
+            String uname = "<i>" + unit.getName() + "</i>";
+            if (unit.isDamaged()) {
                 uname = "<font color='" + MekHQ.getMHQOptions().getFontColorNegativeHexColor() + "'>" + uname
```
Comment (IllianiCBT) [inline MekHQ/src/mekhq/gui/ForceRenderer.java:73]: Here too

**Review by IllianiCBT — APPROVED** [review]
I'm approving this, but I have a couple of comments. Incorporate the changes  onlyif you want, they're non-essential.

**Comment by WeaverThree** [comment]
> I'm approving this, but I have a couple of comments. Incorporate the changes only if you want, they're non-essential.

Will do in a bit

**Comment by WeaverThree** [comment]
Done, and I found a third one :>

**Pull request merged**